secure/storage/src/rocks_db.rs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::rocks_db::database_schema::{SecureStorageKey, SecureStorageSchema, SecureStorageValue};
+use crate::{CryptoKVStorage, Error, GetResponse, KVStorage};
+use anyhow::{anyhow, Result};
+use aptos_crypto::_once_cell::sync::Lazy;
+use aptos_infallible::Mutex;
+use aptos_logger::prelude::*;
+use aptos_time_service::{TimeService, TimeServiceTrait};
+use schemadb::{
+    define_schema,
+    schema::{KeyCodec, ValueCodec},
+    ColumnFamilyName, Options, SchemaBatch, DB,
+};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+use std::{path::Path, sync::Arc, time::Instant};
+
+/// The name of the secure storage db file
+pub const SECURE_STORAGE_DB_NAME: &str = "secure_storage_db";
+
+/// The name of the column family
+const SECURE_STORAGE_CF_NAME: ColumnFamilyName = "secure_storage";
+
+/// We place a global lock around the database to prevent
+/// concurrent threads from creating the same RocksDB file.
+static DATABASE: Lazy<Mutex<Option<Arc<DB>>>> = Lazy::new(|| Mutex::new(None));
```

Comment:
hmm sounds like you want a OnceCell instead?

Reply:
Yeah, I wanted to use it, but the issue is that `reset_and_clear` needs to be able to drop the OnceCell, so it needs to be able to call `take()` after it has been set. But, this requires a mutable global which is unsafe. So, this will have to do for now 😢